controllers/helpers.go

```diff
@@ -301,41 +304,120 @@ func GetSubnetUUID(ctx context.Context, client *prismclientv3.Client, peUUID str
 	return foundSubnetUUID, nil
 }
 
-// GetImage returns an image. If no UUID is provided, returns the unique image with the name.
+// GetImage returns an image UUID. If no UUID is provided, returns the unique image with the name.
 // Returns an error if no image has the UUID, if no image has the name, or more than one image has the name.
-func GetImage(ctx context.Context, client *prismclientv3.Client, id infrav1.NutanixResourceIdentifier) (*prismclientv3.ImageIntentResponse, error) {
+func GetImage(ctx context.Context, client *prismclientv3.Client, id *infrav1.NutanixResourceIdentifier) (*prismclientv3.ImageIntentResponse, error) {
 	switch {
 	case id.IsUUID():
-		resp, err := client.V3.GetImage(ctx, *id.UUID)
+		resolvedUUIDResponse, err := getImageByUUID(ctx, client, *id.UUID)
 		if err != nil {
-			if strings.Contains(fmt.Sprint(err), "ENTITY_NOT_FOUND") {
-				return nil, fmt.Errorf("failed to find image with UUID %s: %v", *id.UUID, err)
-			}
+			return nil, err
 		}
-		return resp, nil
+		return resolvedUUIDResponse, nil
 	case id.IsName():
-		responseImages, err := client.V3.ListAllImage(ctx, "")
+		resolvedUUIDResponse, err := getImageByName(ctx, client, *id.Name)
 		if err != nil {
 			return nil, err
 		}
-		// Validate filtered Images
-		foundImages := make([]*prismclientv3.ImageIntentResponse, 0)
-		for _, s := range responseImages.Entities {
-			imageSpec := s.Spec
-			if strings.EqualFold(*imageSpec.Name, *id.Name) {
-				foundImages = append(foundImages, s)
-			}
+		return resolvedUUIDResponse, nil
+	default:
+	}
```

Comment:
Unless this refactoring is required for the new functionality (and I don't think it is--neither `getImageByUUID` nor `getImageByName` are called by `GetImageWithFormat`), please remove it from this PR.

Reply:
(I think that you were originally refactoring `GetImage` to support lookups, but now that you're moved that out to `GetImageWithFormat`, I think the refactoring is not necessary, and removing it would make it easier for everyone to review the PR)

Reply:
good idea, removed.